basic_bioinf_tools.py

```diff
@@ -0,0 +1,113 @@
+        - "search_for_alt_frames"
+        - "convert_to_nucl_acids"
+        - "three_one_letter_code"
+        - "define_molecular_weight"
+
+    For "search_for_motif" procedure provide:
+    - motif (str): desired motif to check presense in every given sequence\n
+            Example: motif = "GA"
+    - overlapping (bool): count (True) or skip (False) overlapping matches. (Optional)\n
+            Example: overlapping = False
+
+    For "search_for_alt_frames" procedure provide:
+    - alt_start_aa (str): the name of an amino acid that is encoded by alternative start codon (Optional)\n
+            Example: alt_start_aa = "I"
+
+    For "convert_to_nucl_acids" procedure provide:
+    - nucl_acids (str): the nucleic acid to convert to\n
+            Example: nucl_acids = "RNA"\n
+                           nucl_acids = "DNA"\n
+                           nucl_acids = "both"
+
+    Return:
+    - dict: Dictionary with processed sequences. Depends on desired tool\n
+            Please see Readme or desired docstring
+    """
+    procedure_arguments, procedure = protein_tools.check_and_parse_user_input(
+        sequences, **kwargs
+    )
+    return protein_tools.PROTEINS_PROCEDURES_TO_FUNCTIONS[procedure](
+        **procedure_arguments
+    )
+
+
+def run_fastq_filter(
+    seqs: dict[str, tuple[str] | list[str]],
+    gc_bounds: int | float | tuple[int | float] | list[int | float] = (0, 100),
+    length_bounds: int | tuple[int] | list[int] = (0, 2**32),
+    quality_threshold: int | float = 0,
+) -> dict:
+    """
+    Filters a dictionary of DNA sequences and their quality scores based on specified criteria.
+
+    Parameters:
+        seq_dict (Dict[str, Tuple[str, str]]): A dictionary mapping sequence names to tuples containing the DNA sequence
+            as a string and its quality scores as a string.
+        gc_range (Optional[Union[float, Tuple[float, float]]]): A single value or tuple of two values representing
+            the desired range of GC content. If a single value is given, the function will check if the GC content is
+            equal to that value. If a tuple is given, the function will check if the GC content is within the inclusive
+            range of the two values.
+        length_range (Optional[Union[int, Tuple[int, int]]]): A single value or tuple of two values representing the
+            desired range of sequence length. If a single value is given, the function will check if the length of the
+            sequence is equal to that value. If a tuple is given, the function will check if the length of the sequence
+            is within the inclusive range of the two values.
+        quality_threshold (Optional[Union[int, float]]): An integer or float representing the desired quality threshold.
+
+    Returns:
+        Dict[str, Tuple[str, str]]: A dictionary containing only the DNA sequences and their quality scores that pass
+            all specified filters.
+    """
+    fastq_dictionary = {}
```

Comment:
```suggestion
    fastqs = {}
```
1) Не надо в именах переменных указывать их тип данных. 
2) Множественное число